services/httpd/service.go

```diff
@@ -245,10 +267,16 @@ func (s *Service) serveUnixSocket() {
 
 // serve serves the handler from the listener.
 func (s *Service) serve(listener net.Listener) {
+	svr := &http.Server{
```

Comment:
I _think_ that if you just use a single `http.Server` instance for all the listeners, you can just use a single Shutdown call on that.